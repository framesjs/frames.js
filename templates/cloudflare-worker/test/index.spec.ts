// @todo uncomment once @see https://github.com/cloudflare/workers-sdk/issues/5367 is resolved
/* import {
  createExecutionContext,
  waitOnExecutionContext,
  SELF,
} from "cloudflare:test";*/
import { describe, it, expect, vi } from "vitest";
import worker from "../src/index";
import { ExecutionContext } from "@cloudflare/workers-types";

vi.mock("@vercel/og", () => {
  return {
    ImageResponse: class {
      arrayBuffer() {
        return new ArrayBuffer(1);
      }
    },
  };
});

// For now, you'll need to do something like this to get a correctly-typed
// `Request` to pass to `worker.fetch()`.
const IncomingRequest = Request<unknown, IncomingRequestCfProperties<unknown>>;

describe("Hello World worker", () => {
  it("responds with Frame meta tags! (unit style)", async () => {
    const request = new IncomingRequest("http://example.com");
    // Create an empty context to pass to `worker.fetch()`.
    // const ctx = createExecutionContext();
    const ctx = {
      passThroughOnException() {},
      waitUntil(promise) {},
    } satisfies ExecutionContext;
    const response = await worker.fetch(request, { MY_APP_LABEL: "Test" }, ctx);
    // Wait for all `Promise`s passed to `ctx.waitUntil()` to settle before running test assertions
    // await waitOnExecutionContext(ctx);
    expect(await response.text()).toMatchSnapshot();
  });

  /* it("responds with Frame meta tags! (integration style)", async () => {
    const response = await SELF.fetch("https://example.com");
    expect(await response.text()).toMatchSnapshot(`"Hello World!"`);
  });*/
});
