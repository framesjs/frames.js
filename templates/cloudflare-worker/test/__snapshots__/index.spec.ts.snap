// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hello World worker > responds with Frame meta tags! (unit style) 1`] = `"<meta name="og:image" content="data:image/png;base64,AA=="/><meta name="fc:frame" content="vNext"/><meta name="fc:frame:image" content="data:image/png;base64,AA=="/><meta name="fc:frame:post_url" content=""/><meta name="fc:frame:image:aspect_ratio" content="1.91:1"/><meta name="fc:frame:button:1" content="Click Me"/><meta name="fc:frame:button:1:action" content="post"/><meta name="fc:frame:button:1:target" content="http://example.com/?clicked=true&__bi=1-p"/>"`;
